extensions/common/context.cc

```diff
@@ -56,8 +56,13 @@ google::protobuf::util::Status extractNodeMetadata(
   }
   google::protobuf::util::JsonParseOptions json_parse_options;
   json_parse_options.ignore_unknown_fields = true;
-  return JsonStringToMessage(metadata_json_struct, node_info,
-                             json_parse_options);
+  status =
+      JsonStringToMessage(metadata_json_struct, node_info, json_parse_options);
+
+  // used to extract a node key at creation time.
+  absl::StrAppend(node_info->mutable_node_key(), node_info->name(), ".",
+                  node_info->namespace_());
```

Comment:
I wonder why `namespace_` has an underscore but `name` does not. Is this a typo?

Reply:
No protobuf codegen add that.

Reply:
Ah, probably because `namespace` is a keyword.

Reply:
yep.